consensus/state.go

```diff
@@ -278,13 +319,16 @@ func (cs *ConsensusState) startRoutines(maxSteps int) {
 
 func (cs *ConsensusState) OnStop() {
 	cs.QuitService.OnStop()
+	if cs.wal != nil && cs.IsRunning() {
```

Comment:
cs.IsRunning() should always be true, based on QuiteServe/BaseService logic, no?